src/Fixer/LanguageConstruct/NullableTypeDeclarationFixer.php

```diff
@@ -0,0 +1,335 @@
+                ->setAllowedValues([self::OPTION_SYNTAX_UNION, self::OPTION_SYNTAX_QUESTION_MARK])
+                ->setDefault(self::OPTION_SYNTAX_QUESTION_MARK)
+                ->getOption(),
+        ]);
+    }
+
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens): void
+    {
+        $functionsAnalyzer = new FunctionsAnalyzer();
+
+        foreach (array_reverse($this->getElements($tokens), true) as $index => $type) {
+            if ('property' === $type) {
+                $this->normalizePropertyType($tokens, $index);
+
+                continue;
+            }
+
+            // fix return before arguments
+            $this->normalizeMethodReturnType($functionsAnalyzer, $tokens, $index);
+            $this->normalizeMethodArgumentType($functionsAnalyzer, $tokens, $index);
+        }
+    }
+
+    /**
+     * @return array<int, string>
+     *
+     * @phpstan-return array<int, 'method'|'property'>
+     */
+    private function getElements(Tokens $tokens): array
+    {
+        $tokensAnalyzer = new TokensAnalyzer($tokens);
+
+        $elements = array_map(
+            static fn (array $element): string => $element['type'],
+            array_filter(
+                $tokensAnalyzer->getClassyElements(),
+                static fn (array $element): bool => \in_array($element['type'], ['method', 'property'], true)
+            )
+        );
+
+        foreach ($tokens as $index => $token) {
+            if (
+                $token->isGivenKind(T_FN)
+                || ($token->isGivenKind(T_FUNCTION) && !isset($elements[$index]))
+            ) {
+                $elements[$index] = 'method';
+            }
+        }
+
+        return $elements;
+    }
+
+    private function collectTypeAnalysis(Tokens $tokens, int $startIndex, int $endIndex): ?TypeAnalysis
+    {
+        $type = '';
+        $typeStartIndex = $tokens->getNextMeaningfulToken($startIndex);
+        $typeEndIndex = $typeStartIndex;
+
+        for ($i = $typeStartIndex; $i < $endIndex; ++$i) {
+            if ($tokens[$i]->isWhitespace() || $tokens[$i]->isComment()) {
+                continue;
+            }
+
+            $type .= $tokens[$i]->getContent();
+            $typeEndIndex = $i;
+        }
+
+        return '' !== $type ? new TypeAnalysis($type, $typeStartIndex, $typeEndIndex) : null;
+    }
+
+    private function isTypeNormalizable(TypeAnalysis $typeAnalysis): bool
+    {
+        if (!$typeAnalysis->isNullable()) {
+            return false;
+        }
+
+        $type = $typeAnalysis->getName();
+
+        if (str_contains($type, '&')) {
+            return false; // skip DNF types
+        }
+
+        if (!str_contains($type, '|')) {
+            return true;
+        }
+
+        return str_contains($type, '|')
+            && 1 === substr_count($type, '|')
+            && 1 === Preg::match('/(?:\|null$|^null\|)/i', $type);
```

Comment:
```suggestion
        return 1 === substr_count($type, '|') && 1 === Preg::match('/(?:\|null$|^null\|)/i', $type);
```

`str_contains($type, '|')` is already checked above, so it's redundant here.